irods/test/scripts/test007_pam_features_in_new_auth_framework.bats

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bats
+#
+# Test creation of .irodsA for iRODS pam_password authentication, this time purely internal to the PRC
+# library code.
+
+. "$BATS_TEST_DIRNAME"/test_support_functions
+PYTHON=python3
+
+# Setup/prerequisites are same as for login_auth_test.
+# Run as ubuntu user with sudo; python_irodsclient must be installed (in either ~/.local or a virtualenv)
+#
+
+ALICES_ORIGINAL_PAM_PASSWORD=test123
+ALICES_NEW_PASSWORD=test_1234
+
+setup()
+{
+    export SKIP_IINIT_FOR_PASSWORD=1
+    setup_pam_login_for_alice "$ALICES_ORIGINAL_PAM_PASSWORD"
+    unset SKIP_IINIT_FOR_PASSWORD
+}
+
+teardown()
+{
+#    finalize_pam_login_for_alice
+#    test_specific_cleanup
+  :
+}
+
+@test f001 {
+
+    local AUTH_FILE=~/.irods/.irodsA
+
+    # Test assertion: No pre-existing authentication file.
+    [ ! -e $AUTH_FILE ]
+
+    python -c "import sys, irods.client_init; \
+               pw=sys.stdin.readline().strip(); irods.client_init.write_pam_irodsA_file(pw)" \
+              <<<"$ALICES_ORIGINAL_PAM_PASSWORD"
+    [ -e $AUTH_FILE ]
+
+    # ===  test we can log in.
+
+    local SCRIPT="
+import irods.test.helpers as h
+ses = h.make_session()
+try:
+  ses.collections.get(h.home_collection(ses))
+except:
+  print('authenticate error.')
+  exit(120)
+print ('env_auth_scheme=%s' % ses.pool.account._original_authentication_scheme)
+"
+    # Test Python script authenticates.
+    OUTPUT=$($PYTHON -c "$SCRIPT")
+    [[ $OUTPUT = "env_auth_scheme=pam"* ]]
+
+    mc1=$(mtime_and_content $AUTH_FILE)
```

Comment:
What does `mtime_and_content` return? Specifically, the `_and_content` part of it.

Reply:
iIt's just a way of checking a file was not touched. it checks both mtime and the file contents to verify this.

Reply:
How does it verify that? What is it comparing it to?

Reply:
if we do:
```
mc1=$(mtime_and_content $FILE1)
mc2=$(mtime_and_content $FILE2)
```
then [ "$mc1" = "$mc2" ] will verify the two files as having the same mtime and the same file content.
`mtime_and_content` is a Bash function that simply produces a concatenated mtime stamp, "\n", then followed by the acutal file content.

Reply:
Got it.